services/jitpack/jitpack-downloads.service.js

```diff
@@ -1,14 +1,83 @@
 'use strict'
 
-const { deprecatedService } = require('..')
+const Joi = require('joi')
+const { nonNegativeInteger } = require('../validators')
+const { BaseJsonService } = require('..')
+const { downloadCount } = require('../color-formatters')
+const { metric } = require('../text-formatters')
 
-module.exports = deprecatedService({
-  route: {
-    base: 'jitpack',
-    pattern: ':period(dw|dm)/:various*',
+const schema = Joi.object({
+  week: nonNegativeInteger,
+  month: nonNegativeInteger,
+}).required()
+
+const variantMap = {
+  dw: {
+    api_field: 'week',
+    suffix: '/week',
+  },
+  dm: {
+    api_field: 'month',
+    suffix: '/month',
   },
-  message: 'temporarily unavailable',
-  label: 'downloads',
-  category: 'downloads',
-  dateAdded: new Date('2019-05-27'),
-})
+}
+
+module.exports = class JitpackDownloads extends BaseJsonService {
+  static get category() {
+    return 'downloads'
+  }
+
+  static get route() {
+    return {
+      base: 'jitpack',
+      pattern:
+        ':variant(dw|dm)/:vcs(github|bitbucket|gitlab|gitee)/:user/:repo',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'JitPack - Downloads',
+        namedParams: {
+          variant: 'dm',
+          vcs: 'github',
+          user: 'jitpack',
+          repo: 'maven-simple',
+        },
+        staticPreview: JitpackDownloads.render({
+          downloads: 14000,
+          variant: 'dm',
+        }),
+        keywords: ['java', 'maven'],
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'downloads' }
+  }
+
+  static render({ downloads, variant }) {
+    return {
+      message: `${metric(downloads)}${variantMap[variant].suffix}`,
+      color: downloadCount(downloads),
+    }
+  }
+
+  async fetch({ vcs, user, repo }) {
+    return this._requestJson({
+      schema,
+      url: `https://jitpack.io/api/downloads/com.${vcs}.${user}/${repo}`,
+      errorMessages: { 401: 'project not found or private' },
```

Comment:
One downside of this new API is that it doesn't return a 404/401 response when projects are not found like the previous (albeit unusable API) did, this one just returns a value of `0` for downloads. Keeping this in here though in case this new API adds 401/404 responses in the future